src/Abstractions/src/Asp.Versioning.Abstractions/DeprecationPolicy.cs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+
+namespace Asp.Versioning;
+
+/// <summary>
+/// Represents an API version deprecation policy.
+/// </summary>
+public class DeprecationPolicy
+{
+    private readonly LinkList links;
```

Comment:
This should be:

```c#
private readonly DeprecationLinkList links;
```

so that it can be _devirtualized_ and optimized.

I this _should_ also initially be `null` internally so that calls do not pay the cost of allocation if the policy doesn't add any links. This would be a similar pattern used in `SunsetPolicy`. That is also the reason for the `HasLinks` property. `Links.Count == 0` could have just as easily worked externally, but then it always requires an allocation just to find out it's empty.